packages/reactivity/src/ref.ts

```diff
@@ -66,9 +66,6 @@ export function isRef(r: any): r is Ref {
   return Boolean(r && r.__v_isRef === true)
 }
 
-export function ref<T extends object>(
-  value: T
-): [T] extends [Ref] ? T : Ref<UnwrapRef<T>>
```

Comment:
I don't see the use case for this overload. Am I missing something here?

Reply:
Looks like this was added in #4734.
Should be save to remove as the added test still passes.